frontend/packages/console-plugin-sdk/src/typings/perspective.ts

```diff
@@ -0,0 +1,24 @@
+import { Extension } from '.';
+
+namespace ExtensionProperties {
+  export interface Perspective {
+    /* The perspective identifier. */
+    id: string;
+    /* The perspective display name. */
+    name: string;
+    /* The perspective display icon. */
+    icon: React.ReactNode;
+    /* The perspective landing page URL. */
+    landingPageURL: string;
+    /* Whether the perspective is the default. There can only be one default. */
```

Comment:
What happens if:

- a plugin declares two perspectives each with `default: true`
- two plugins each declare a perspective with `default: true`

This could be handled by taking the first default one and discarding the rest, or just omitting the `default` option and assume there will always be at least one perspective (added by Console application itself) and that one is the default (because it's declared as the first one).

@christianvogt @spadgett What are your thoughts on this?

This relates to my comment above, regarding plugin evaluation order, i.e. Console application's plugin should be processed before any other plugins.

Also, what happens if a plugin tries to redefine existing perspective (same `id`)?

Reply:
I think there should be a generic mechanism for each extension owner (if we ever support plugins providing their own extensions then we'll have more than console core owning extensions) to define their own extension processor. This would allow the owner to validate the contributed extensions. Duplicate perspective IDs or multiple defined as a `default` would be runtime errors.

Maybe we should create an `ExtensionProcessor` extension :) wherein you contribute a function to process extensions by `type`. The function would receive the list of contributed extensions and it could throw an error in case of duplicate IDs. It could also log warnings. The return value is an array of extensions in case the processor chose to add or subtract from the list. This mechanism could be used to contribute CRD based extensions in the future. eg a mega-menu extension will get extra values contributed via CRDs.